llvm/lib/Target/AArch64/AArch64LoopIdiomTransform.cpp

```diff
@@ -202,32 +227,135 @@ bool AArch64LoopIdiomTransform::run(Loop *L) {
                     << CurLoop->getHeader()->getParent()->getName()
                     << "] Loop %" << CurLoop->getHeader()->getName() << "\n");
 
-  return recognizeByteCompare();
+  MemCompareIdiom BCI(TTI, DT, LI, L);
+  if (BCI.recognize()) {
```

Comment:
I wanted to leave the door open for recognising more types than just bytes. Also, given the class itself is called `MemCompareIdiom` I was worried that `MemCompareIdiom::recognizeByteCompare` was a bit wordy, but I'm happy to  change it to `recognizeMemCompare` instead of just `recognize` if that helps?